mobile/src/home-view.js

```diff
@@ -269,13 +269,14 @@ class HomeView extends Component {
           return a.order - b.order
         })
         this.originalOrder(otherQuestions)
-        var orderedQuestions = pinnedQuestions.concat(otherQuestions)
-        if (currentSort === "Approved" || "Popular" || "Recent" && orderedQuestions.length) { 
-          orderedQuestions = orderedQuestions.filter(item => item.block === false && item.answered === false && item.approve && item.new === false && item.session === session.key)
+        const allQuestions = pinnedQuestions.concat(otherQuestions)
+        let orderedQuestions = allQuestions
+        if (currentSort === "Approved" || "Popular" || "Recent" && allQuestions.length) { 
```

Comment:
This is likely not what you are intending.

1. `&&` has a higher operator precedence than `||` so this will be interpreted the same as: `currentSort === "Approved" || "Popular" || ("Recent" && allQuestions.length)`
2. `if (a === b || c)` does not mean "if a is the same as either b or c".  It means "if a is the same as b, or c is truthy"

Reply:
What is the best way to handle cases like this?